projects/apache-httpd/fuzz_utils.c

```diff
@@ -146,8 +150,9 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
       apr_interval_time_t timeout;
       ap_timeout_parameter_parse(new_str, &timeout, "ms");
 
-      tmp_s = new_str;
-      ap_content_type_tolower(tmp_s);
+      tmp_s_nc = strdup(new_str);
```

Comment:
So I pushed a (larger) commit to grab random bytes anytime a function modifies its input, such that the modifications do no influence the next fuzzing (e.g. url-encoding then continuing with the modified string would quite limit the charset for the next fuzzed function).

Though it requires that there are enough random bytes to begin with or `af_gb_get_null_terminated()` will return `NULL` for the latest tests...

Reply:
Sounds good, let's try this out. After running for a bit the fuzzer should identify that larger inputs trigger more code, so eventually it will get there. I will keep an eye on coverage to ensure all API calls happen - we can always put a condition early in the fuzzer function that checks the size of the input, which forces the fuzzer to quickly generate large inputs.